src/bindgen/parser.rs

```diff
@@ -265,16 +265,25 @@ impl<'a> Parser<'a> {
             self.cache_src.get(&owned_mod_path).unwrap().clone()
         };
 
-        let mod_dir = mod_path.parent().unwrap();
+        // Compute module directory according to Rust 2018 rules
+        let mod_dir_2018;
 
-        self.process_mod(pkg, mod_dir, &mod_parsed)
+        let mod_dir = if depth == 0 || mod_path.ends_with("mod.rs") {
```

Comment:
Hmm, I still don't think this is quite sound is it? At the very least we should check the file name for full equality for `mod.rs`, not just whether the path ends with `mod.rs`

Reply:
Apparently `Path::ends_with` only checks entire path components so it is checking if the (entire) file name is `mod.rs`. This wouldn't be the case with a `&str`.

https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.ends_with